src/mesh/Router.cpp

```diff
@@ -624,6 +630,12 @@ void Router::handleReceived(meshtastic_MeshPacket *p, RxSource src)
         // After potentially altering it, publish received message to MQTT if we're not the original transmitter of the packet
         if ((decoded || p_encrypted->pki_encrypted) && moduleConfig.mqtt.enabled && !isFromUs(p) && mqtt)
             mqtt->onSend(*p_encrypted, *p, p->channel);
+#endif
+#if HAS_UDP_MULTICAST
+        if ((decoded || p_encrypted->pki_encrypted) && !isFromUs(p) && udpThread &&
```

Comment:
I think this logic should be different than for MQTT. The reason for MQTT it is also sending everything it receives is because MQTT is used not only for transport (bridging) but also as client. So a packet directed towards you, which is not rebroadcasted, also ends up on the MQTT server. For UDP I think we should only use `onSend()` when the Router sends, and thus remove the `isFromUs()` there.

Reply:
I was thinking about this as well. It makes sense to remove the isFromUs check

Reply:
I meant to remove `isFromUs(p)` from `Router::send()` and remove the complete block out of `Router::handleReceived()` (since packets received via UDP already arrive here). This way it only sends via UDP if it would via LoRa.
However, if you want it to always rebroadcast over UDP (without decrementing the hop limit, before adding yourself to the traceroute, etc.) then we could do it the same as for MQTT. So depends on whether we see this as a normal hop or more a virtual one.